frontend/app/src/components/asset-manager/AssetManagement.vue

```diff
@@ -86,9 +108,12 @@ import { assert } from '@/utils/assertions';
   components: { MergeDialog, ConfirmDialog, AssetForm, BigDialog, AssetTable },
   computed: {
     ...mapState('balances', ['supportedAssets'])
+  },
+  methods: {
+    ...mapActions('session', ['logout'])
   }
 })
-export default class AssetManagement extends Vue {
+export default class AssetManagement extends Mixins(RestoreAssetsDatabase) {
```

Comment:
`RestoreAssetsDatabase` doesn't seem like a mixing, it is a regular component I there any reason why `AssetManagement` should extend it instead of using it in the template?

Reply:
Okey this I had doubts. I believed the right way to extend another component was using a Mixin by looking at other examples. The reason is that the methods I had defined in `RestoreAssetsDatabase` are used here too in order to execute the _soft_ reset. `AssetManagement` hold the UI for the soft reset and `RestoreAssetsDatabase` the UI for the _hard_ one.

Should I've made instead
```suggestion
export default class AssetManagement extends RestoreAssetsDatabase {
```

Reply:
A mixin is a special case, not a normal component. If you check the code you can see that all the mixins are pure typescript classes without any template. If it is a normal component the normal extend should work. 

Alternatively, you can extract the common code to a mixin class and then have both components extend the same mixin. 

These mixins will probably go away when we migrate to the composition API though since it provides the means to introduce self-contained pieces of code use can re-use that are better than mixins.